lib/humanized_id/humanizer.rb

```diff
@@ -1,43 +1,38 @@
 # frozen_string_literal: true
 module HumanizedId
   class Humanizer
-    SOURCE_CHARSET = %w(0123456789abcdefghijklmnopqrstuvwxyz).freeze
+    SOURCE_CHARSET = (('0'..'9').to_a + ('a'..'z').to_a).freeze
 
     def initialize(id:, min_length: nil, prefix: '')
       @id               = id
       @min_length       = min_length
-      @prefix           = prefix.nil? ? '' : prefix
-      @target_charset   = HumanizedId::CHARACTERSET.join('')
-      @source_charset   = SOURCE_CHARSET.join('')
+      @prefix           = prefix.to_s
+      @target_charset   = CHARSET.join
+      @source_charset   = SOURCE_CHARSET.join
     end
 
     def generate_humanized_id
-      new_id = convert_to_target_base id: @id
-      new_id = convert_to_target_charset id: new_id
-      new_id = resize id: new_id if @min_length
+      commands = %i(convert_to_target_base convert_to_target_charset resize)
```

Comment:
While I can understand your reasoning @msmirnoff , I think that there is a legitimate case to identify overwriting the same variable with a new value produced by a computation on its previous value as a code smell. If that variable were an attribute, thus holding state, I'd definitely point at it as an issue.

Ruby doesn't natively support functional composition, but there are ways to get to it, like the above.